resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -259,9 +259,9 @@ private[spark] object KubernetesConf {
     s"$appName-$id"
       .trim
       .toLowerCase(Locale.ROOT)
-      .replaceAll("[^a-z0-9\\-]", "-")
+      .replaceAll("[^a-z0-9\\-]", "x")
       .replaceAll("-+", "-")
-      .replaceAll("^-", "")
+      .replaceAll("^[0-9\\-]", "x")
```

Comment:
Let's keep the original one and have like the following.
```scala
.replaceAll("^-", "")
.replaceAll("^[0-9]", "x")
```

Reply:
updated as suggested